build-common/NHibernate.props

```diff
@@ -6,7 +6,7 @@
     <VersionPatch  Condition="'$(VersionPatch)'  == ''">0</VersionPatch>
     <!-- Clear VersionSuffix for making release and set it to dev for making development builds -->
     <VersionSuffix Condition="'$(VersionSuffix)' == ''">dev</VersionSuffix>
-    <LangVersion Condition="'$(MSBuildProjectExtension)' != '.vbproj'">12.0</LangVersion>
+    <LangVersion Condition="'$(MSBuildProjectExtension)' != '.vbproj'">preview</LangVersion>
```

Comment:
So, it needs language version 14.0? We will likely wait for it before merging your proposal, if we accept it.

Reply:
I have not spotted what does require such a change. Why this change?

Reply:
It's 13.0 as such, and comes with .NET 9.0, but for now you need to set the LangVersion to preview on .NET 9.0 in order to lock on a System.Threading.Lock object. The expectation is that this changes as soon as .NET 9.0 is released.